prow/tide/status.go

```diff
@@ -360,19 +366,79 @@ func (sc *statusController) sync(pool map[string]PullRequest) {
+	sc.setStatuses(sc.search(), pool)
+}
+
+func (sc *statusController) search() []PullRequest {
+	orgs, repos := sc.ca.Config().Tide.Queries.OrgsAndRepos()
+	freshOrgs, freshRepos := orgs.Difference(sc.trackedOrgs), repos.Difference(sc.trackedRepos)
+	oldOrgs, oldRepos := sc.trackedOrgs.Difference(orgs), sc.trackedRepos.Difference(repos)
+	// Stop tracking orgs and repos that aren't queried this loop.
+	sc.trackedOrgs.Delete(oldOrgs.UnsortedList()...)
+	sc.trackedRepos.Delete(oldRepos.UnsortedList()...)
+	// Determine the query for tracked PRs now before we modify 'trackedOrgs' and 'trackedRepos'.
+	var trackedQuery string
+	if sc.trackedOrgs.Len() > 0 || sc.trackedRepos.Len() > 0 {
+		trackedQuery = openPRsQuery(sc.trackedOrgs.UnsortedList(), sc.trackedRepos.UnsortedList())
+	}
+	queryStartTime := time.Now()
+
+	var lock sync.Mutex
+	var wg sync.WaitGroup
+	var allPRs []PullRequest
+	// Query fresh orgs and repos individually and since the beginning of time.
+	// These queries are larger and more likely to fail so we query for targets individually.
+	singleTargetSearch := func(query, target string, tracked sets.String) {
+		defer wg.Done()
+		searcher := newSearchExecutor(context.Background(), sc.ghc, sc.logger, query)
+		prs, err := searcher.search()
+		if err != nil {
+			sc.logger.WithError(err).Errorf("Searching for open PRs in %s.", target)
+			return
+		}
+		func() {
+			lock.Lock()
+			defer lock.Unlock()
+			allPRs = append(allPRs, prs...)
+			tracked.Insert(target)
+		}()
+	}
+
+	wg.Add(freshOrgs.Len() + freshRepos.Len())
+	for _, org := range freshOrgs.UnsortedList() {
+		go singleTargetSearch(openPRsQuery([]string{org}, nil), org, sc.trackedOrgs)
+	}
+	for _, repo := range freshRepos.UnsortedList() {
+		go singleTargetSearch(openPRsQuery(nil, []string{repo}), repo, sc.trackedRepos)
+	}
+	wg.Wait()
+
+	// Query tracked orgs and repos together and only since the last time we queried.
 	// We offset for 30 seconds of overlap because GitHub sometimes doesn't
 	// include recently changed/new PRs in the query results.
-	sinceTime := sc.lastSuccessfulQueryStart.Add(-30 * time.Second)
-	query := sc.ca.Config().Tide.Queries.AllOpenPRs()
-	queryStartTime := time.Now()
-	searcher := newSearchExecutor(context.Background(), sc.ghc, sc.logger, query)
-	allPRs, err := searcher.searchSince(sinceTime)
-	if err != nil {
-		sc.logger.WithError(err).Errorf("Searching for open PRs.")
-		return
+	if trackedQuery != "" {
+		sinceTime := sc.lastSuccessfulQueryStart.Add(-30 * time.Second)
+		searcher := newSearchExecutor(context.Background(), sc.ghc, sc.logger, trackedQuery)
+		prs, err := searcher.searchSince(sinceTime)
+		if err != nil {
+			sc.logger.WithError(err).Error("Searching for open PRs from 'tracked' orgs and repos.")
+		} else {
+			allPRs = append(allPRs, prs...)
+		}
 	}
+
 	// We were able to find all open PRs so update the last successful query time.
 	sc.lastSuccessfulQueryStart = queryStartTime
-	sc.setStatuses(allPRs, pool)
```

Comment:
Why did we lose this?

Reply:
it's above: https://github.com/kubernetes/test-infra/pull/9326/files#diff-533bbd39f81d8223a55b40074200ca4cR369

Reply:
Yeah, the diff is just a bit confusing.

Reply:
Ah. Changes LGTM